### PR TITLE
adding searchIngredients

### PR DESCRIPTION
**Pull request by suzannealexander**
adding basic functionality for searching for ingredients

### CODE REVIEW

File: src/routes/search.py [hunk 1]
```diff
@@ -25,17 +25,25 @@ def get_blueprint():
     return search_blueprint
 
 
-@search_blueprint.route("/search_ingredients")
+
```
Comment (Sedwards8900) [inline src/routes/search.py:29]: A brief description of the route's functionality is needed

File: src/routes/search.py [hunk 2]
```diff
@@ -25,17 +25,25 @@ def get_blueprint():
...
+@search_blueprint.route("/search_ingredients", methods=["POST"])
 def search_ingredients():
-    return render_template("search.html")
+    if flask.request.method == "POST":
+        ingredients = flask.request.form.get("searchIngredients")
```
Comment (Sedwards8900) [inline src/routes/search.py:33]: I also recommend "import flask" or "from flask import request" at the beginning of the file to reduce calling the full library each time

File: src/routes/search.py [hunk 3]
```diff
@@ -25,17 +25,25 @@ def get_blueprint():
...
+        # prints what the user searches for to the terminal
+        print(ingredients)
+        returnedDict = api.spoonacular.search_ingredients(ingredients)
```
Comment (Sedwards8900) [inline src/routes/search.py:36]: The top comment "import spoonacular somehow" is too generic and needs to be updated with "from api.spoonacular import search_ingredients"

File: src/routes/search.py [hunk 4]
```diff
@@ -25,17 +25,25 @@ def get_blueprint():
...
+        # prints the 10 recipes to the terminal
+        print(returnedDict)
+        return render_template("index.html", ingredients=returnedDict)
+    return render_template("index.html")
 
 
 @search_blueprint.route("/search_recipes", methods=["POST"])
```
Comment (Sedwards8900) [inline src/routes/search.py:43]: Brief description of route must be added as well

File: src/routes/search.py [hunk 5]
```diff
@@ -25,17 +25,25 @@ def get_blueprint():
...
 def search_recipes():
     if flask.request.method == "POST":
```
Comment (Sedwards8900) [inline src/routes/search.py:45]: See comment above regarding importing flask. Update accordingly throughout the file.

File: src/routes/search.py [hunk 6]
```diff
@@ -25,17 +25,25 @@ def get_blueprint():
...
         ingredients = flask.request.form.get("searchRecipes")
         # prints what the user searches for to the terminal
-        #
         print(ingredients)
         returnedDict = api.spoonacular.search_recipes(ingredients)
```
Comment (Sedwards8900) [inline src/routes/search.py:49]: See above comment regarding import api.spoonacular and update accordingly throughout the file

File: templates/index.html [hunk 7]
```diff
@@ -66,8 +66,24 @@ <h2>{{recipe['name']}}</h2>
         <!-- <h2>Sushi</h2>
```
Comment (Sedwards8900) [inline templates/index.html:66]: Remove temporary/sample text once integration of search functionality results into index.html is complete

**Review by Sedwards8900 — CHANGES_REQUESTED** [review]
Hi Suzanne, please refer to the comments regarding import libraries and files within the project as well as adding commentary as to the functionality of the routes added

File: src/routes/search.py (hunk 5 above)
Comment (suzannealexander) [inline src/routes/search.py:45]: ``
```suggestion
    if  request.method == "POST":
```

File: src/routes/search.py [hunk 8]
```diff
@@ -25,17 +25,25 @@ def get_blueprint():
...
-    return render_template("search.html")
+    if flask.request.method == "POST":
+        ingredients = flask.request.form.get("searchIngredients")
+        # prints what the user searches for to the terminal
+        print(ingredients)
+        returnedDict = api.spoonacular.search_ingredients(ingredients)
+        # prints the 10 recipes to the terminal
+        print(returnedDict)
+        return render_template("index.html", ingredients=returnedDict)
+    return render_template("index.html")
```
Comment (suzannealexander) [inline src/routes/search.py:42]:
```suggestion
#route allowing users to search for recipes by ingredient, returning image, name of dish, and recipe id
```

File: src/routes/search.py [hunk 9]
```diff
@@ -25,17 +25,25 @@ def get_blueprint():
     return search_blueprint
```
Comment (suzannealexander) [inline src/routes/search.py:28]:
```suggestion
#route for users to search ingredients and retrieve their name, picture and ingredient id
```

File: src/routes/search.py [hunk 10]
```diff
@@ -25,17 +25,25 @@ def get_blueprint():
...
 
 
-@search_blueprint.route("/search_ingredients")
+
+@search_blueprint.route("/search_ingredients", methods=["POST"])
 def search_ingredients():
-    return render_template("search.html")
+    if flask.request.method == "POST":
```
Comment (suzannealexander) [inline src/routes/search.py:32]:
```suggestion
    if request.method == "POST":
```

File: src/routes/search.py (hunk 2 above)
Comment (suzannealexander) [inline src/routes/search.py:33]:
```suggestion
        ingredients = request.form.get("searchIngredients")
```

File: src/routes/search.py (hunk 5 above)
Comment (suzannealexander) [inline src/routes/search.py:45]:
```suggestion
    if request.method == "POST":
```

File: src/routes/search.py [hunk 11]
```diff
@@ -25,17 +25,25 @@ def get_blueprint():
...
+        # prints the 10 recipes to the terminal
+        print(returnedDict)
+        return render_template("index.html", ingredients=returnedDict)
+    return render_template("index.html")
 
 
 @search_blueprint.route("/search_recipes", methods=["POST"])
 def search_recipes():
     if flask.request.method == "POST":
         ingredients = flask.request.form.get("searchRecipes")
```
Comment (suzannealexander) [inline src/routes/search.py:46]:
```suggestion
        ingredients = request.form.get("searchRecipes")
```

File: src/routes/search.py [hunk 12]
```diff
@@ -25,17 +25,25 @@ def get_blueprint():
     return search_blueprint
 
 
-@search_blueprint.route("/search_ingredients")
+
+@search_blueprint.route("/search_ingredients", methods=["POST"])
 def search_ingredients():
```
Comment (suzannealexander) [inline src/routes/search.py:31]:
```suggestion
def searchByingredients():
```

File: src/routes/search.py (hunk 3 above)
Comment (suzannealexander) [inline src/routes/search.py:36]:
```suggestion
        returnedDict = search_ingredients(ingredients)
```

File: src/routes/search.py (hunk 6 above)
Comment (suzannealexander) [inline src/routes/search.py:49]:
```suggestion
        returnedDict = search_recipes(ingredients)
```

File: src/routes/search.py [hunk 13]
```diff
@@ -25,17 +25,25 @@ def get_blueprint():
...
+        print(ingredients)
+        returnedDict = api.spoonacular.search_ingredients(ingredients)
+        # prints the 10 recipes to the terminal
+        print(returnedDict)
+        return render_template("index.html", ingredients=returnedDict)
+    return render_template("index.html")
 
 
 @search_blueprint.route("/search_recipes", methods=["POST"])
 def search_recipes():
```
Comment (suzannealexander) [inline src/routes/search.py:44]:
```suggestion
def searchByrecipes():
```

File: src/routes/search.py (hunk 5 above)
Comment (suzannealexander) [inline src/routes/search.py:45]:
```suggestion
    if request.method == "POST":
```

File: templates/index.html (hunk 7 above)
Comment (suzannealexander) [reply templates/index.html:66]: will implement these specific changes in next push!

**Review by Sedwards8900 — APPROVED** [review]

**Pull request merged**